### ISSUE TITLE
`unit-no-unknown` not ignore hacked unit

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#1645]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

`unit-no-unknown` not ignore hacked unit

> Which rule, if any, is this issue related to?

e.g. `unit-no-unknown`

> What CSS is needed to reproduce this issue?

e.g.

```css
 @media screen and (min-width:0\0) {
        img {
            height: 100%; // svg height fix in ie
        }
    }
```

> What stylelint configuration is needed to reproduce this issue?

No matter

> Which version of stylelint are you using?

e.g. `7.0.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

No matter

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No matter

> What did you expect to happen?

No warnings to be flagged

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
test.css
✖  Unexpected unknown unit "\0"   unit-no-unknown
```
